code/game/objects/structures/signs/_signs.dm

```diff
@@ -8,6 +10,25 @@
 	armor = list("melee" = 50, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50, "stamina" = 0)
 	var/buildable_sign = 1 //unwrenchable and modifiable
 	rad_flags = RAD_PROTECT_CONTENTS | RAD_NO_CONTAMINATE
+	var/random_dept_base  // will be replaced by a random subtype if the station has random_dept and this is defined
+
+/obj/structure/sign/Initialize(mapload)
+	. = ..()
+	if(random_dept_base)
+		GLOB.dept_signs += src
+
+/obj/structure/sign/proc/randomize()
+	if(!random_dept_base)
+		return
+	var/list/all_dept = subtypesof(random_dept_base)
+	while(TRUE)
```

Comment:
Is while(TURE) necessary?

Reply:
its so it keeps trying in case it gets one with no defined iconstate (since that would look weird)

Reply:
In my opinion, var/obj/structure/sign/departments/new_dept should be outside of the while, and the while should have the if below.

Reply:
?? but it wont pick a different sprite if you put the new_dept var outside the while

Reply:
I am 100% certain that there is a way to work the same code you wrote with putting the var outside. While(TRUE) there doesn't look ideal.

Reply:
The point is not to use while(TURE)